includes/rest-api/class-sensei-rest-api-setup-wizard-controller.php

```diff
@@ -0,0 +1,329 @@
+		$this->setupwizard = Sensei_Onboarding::instance();
+	}
+
+	/**
+	 * Register the REST API endpoints for Setup Wizard.
+	 */
+	public function register_routes() {
+
+		$this->register_progress_route();
+		$this->register_welcome_route();
+		$this->register_purpose_route();
+		$this->register_features_route();
+	}
+
+	/**
+	 * Check user permission for REST API access.
+	 *
+	 * @return bool Whether the user can access the Setup Wizard REST API.
+	 */
+	public function can_user_access_rest_api() {
+		return current_user_can( 'manage_sensei' );
+	}
+
+	/**
+	 * Register /welcome endpoint.
+	 */
+	public function register_welcome_route() {
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base . '/welcome',
+			[
+				[
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => [ $this, 'submit_welcome' ],
+					'permission_callback' => [ $this, 'can_user_access_rest_api' ],
+					'args'                => [
+						'usage_tracking' => [
+							'required' => true,
+							'type'     => 'boolean',
+						],
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Register /purpose endpoint.
+	 */
+	public function register_purpose_route() {
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base . '/purpose',
+			[
+				[
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => [ $this, 'submit_purpose' ],
+					'permission_callback' => [ $this, 'can_user_access_rest_api' ],
+					'args'                => [
+						'selected' => [
+							'required' => true,
+							'type'     => 'array',
+							'items'    => [
+								'type' => 'string',
+								'enum' => self::PURPOSES,
+							],
+						],
+						'other'    => [
+							'required' => true,
+							'type'     => 'string',
+						],
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Register /features endpoint.
+	 */
+	public function register_features_route() {
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base . '/features',
+			[
+				[
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => [ $this, 'submit_features' ],
+					'permission_callback' => [ $this, 'can_user_access_rest_api' ],
+					'args'                => [
+						'selected' => [
+							'required' => true,
+							'type'     => 'array',
+							'items'    => [
+								'type' => 'string',
+								'enum' => $this->setupwizard->plugin_slugs,
+							],
+						],
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Register GET / endpoint.
+	 */
+	public function register_progress_route() {
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base,
+			[
+				[
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => [ $this, 'get_data' ],
+					'permission_callback' => [ $this, 'can_user_access_rest_api' ],
+				],
+				'schema' => [ $this, 'get_schema' ],
+			]
+		);
+	}
+
+	/**
+	 * Get data for Setup Wizard frontend.
+	 *
+	 * @return mixed Setup Wizard data
```

Comment:
Maybe should it be an `array` instead of `mixed`?

Reply:
Fixed in 96c1d7dd